### ISSUE TITLE
prefer-primordials: Cache prototype for instance check

### ISSUE THREAD

**Issue by petamoriken** [denoland/deno_lint#986]
### Lint Name

prefer-primordials

### Code Snippet

```ts
a instanceof b;
b[Symbol.hasinStance](a);
```

### Expected Result

Suggest using `ObjectPrototypeIsPrototypeOf`.

```ts
ObjectPrototypeIsPrototypeOf(bPrototype, a);
```

### Actual Result

Nothing.

### Version

deno 1.16.4 (release, aarch64-apple-darwin)
v8 9.7.106.15
typescript 4.4.2

**Comment by lucacasonato** [comment]
Oh why is JS like this. :(

**Comment by petamoriken** [comment]
The same may be said for `in` 👉 `ObjectHasOwn`.
